source/common/http/codec_client.cc

```diff
@@ -72,16 +72,16 @@ void CodecClient::deleteRequest(ActiveRequest& request) {
 }
 
 RequestEncoder& CodecClient::newStream(ResponseDecoder& response_decoder) {
-  ActiveRequestPtr request(new ActiveRequest(*this, response_decoder));
-  request->encoder_ = &codec_->newStream(*request);
-  request->encoder_->getStream().addCallbacks(*request);
+  ActiveRequestPtr request(
```

Comment:
If this is just based on protocol, can't we switch on CodecType here and skip the new interface entirely?

I think we should also comment (here or if you stick with the interface, there) why we skip this for HTTP/1.1

Reply:
OK, we can do that to keep it simpler.